src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -422,6 +422,33 @@ def remove_subvolume(self, fs_handle, **kwargs):
             ret = self.volume_exception_to_retval(ve)
         return ret
 
+    @connection_pool_wrap
+    def resize_subvolume(self, fs_handle, **kwargs):
+        ret        = 0, "", ""
+        subvolname = kwargs['sub_name']
+        newsize    = kwargs['new_size']
+        groupname  = kwargs['group_name']
+        limitshrink= kwargs['limit_shrink']
+
+        try:
+            with SubVolume(self.mgr, fs_handle) as sv:
+                spec = SubvolumeSpec(subvolname, groupname)
+                if not self.group_exists(sv, spec):
+                    raise VolumeException(
+                        -errno.ENOENT, "Subvolume group '{0}' not found, create it with " \
+                        "'ceph fs subvolumegroup create' before creating or resizing subvolumes".format(groupname))
+                subvolpath = sv.get_subvolume_path(spec)
+                if not subvolpath:
+                    raise VolumeException(
+                        -errno.ENOENT, "Subvolume '{0}' not found, create it with " \
+                        "'ceph fs subvolume create' before resizing subvolumes".format(subvolname))
+                nsize, usedbytes = sv.resize_subvolume(subvolpath, newsize, limitshrink)
+                ret = 0, json.dumps([{'UsedInBytes': usedbytes},{'QuotaInBytes': nsize},
+                                     {'PercentageUsed': '{0:.2f}'.format((float(usedbytes) / nsize) * 100.0)}], indent=2), ""
```

Comment:
JSON is usually lowercase and `_` between words.

Let's use: `{'bytes_used': XXX, 'bytes_quota': YYY, 'bytes_pcent': ZZ.ZZ}`

Reply:
Also: `newsize` may be 0. That would disable the quota. We do want to allow that but then you would be dividing by 0.

I think we'd need to do `statfs` on the file system to see how much space is available. It needs a separate test case.

Reply:
> Also: `newsize` may be 0. That would disable the quota. We do want to allow that but then you would be dividing by 0.
> 

Also it doesn't make sense to display a percentage value when the numerator is higher than the denominator (Not only when the denominator is 0) in the percentage calculation. In such case, I think it should be always 100%?